PxWeb/Filters/Api2/LangValidationFilter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace PxWeb.Filters.Api2
+{
+    public class LangValidationFilter : Attribute, IResourceFilter
+    {
+        private readonly List<string> _languages;
+
+        public LangValidationFilter(List<string> langs)
+        {
+            _languages = langs;
+        }
+
+        public void OnResourceExecuting(ResourceExecutingContext context)
+        {
+            var lanValues = context.HttpContext.Request.Query["lang"].ToString();
+
+            if (!string.IsNullOrEmpty(lanValues) && !_languages.Exists(x => x.ToString() == lanValues))
+            {
+                //context.Result = new CustomResult("No such language" );
+                context.Result = new NotFoundResult();
```

Comment:
@KentMossback you should change it to a BadRequest similar to `BadRequestObjectResult($"language {lanValues} is not a valid language");` instead of a `NotFoundResult`.

Reply:
Changed code to your suggestion